manager/orchestrator/globaljob/orchestrator.go

```diff
@@ -2,19 +2,18 @@ package globaljob
 
 import (
 	"context"
-	"fmt"
 	"sync"
 
-	"github.com/docker/swarmkit/api"
-	"github.com/docker/swarmkit/manager/constraint"
-	"github.com/docker/swarmkit/manager/orchestrator"
 	"github.com/docker/swarmkit/manager/state/store"
-	gogotypes "github.com/gogo/protobuf/types"
 )
 
 type Orchestrator struct {
 	store *store.MemoryStore
 
+	// rec is used to decouple reconciliation logic from event loop logic, and
+	// allows injecting a fake reconciler.
```

Comment:
Looks like some dependency injection could be of use here, something like https://github.com/google/wire ? Just food for thought, way out of the scope of this change :)